### PR TITLE
fix: Add missing types on Shipment object

### PR DESCRIPTION
**Pull request by clalexander**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change. (None needed)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [Shippo Shipments](https://docs.goshippo.com/shippoapi/public-api/#tag/Shipments)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

Related to issue #69261 and discussion #69182

### CODE REVIEW

**Comment by clalexander** [comment]
> Hey @clalexander,
> 
> 😒 Your PR doesn't modify any tests, so it's hard to know what's being fixed, and your changes might regress in the future. Please consider [adding tests](https://github.com/DefinitelyTyped/DefinitelyTyped#user-content-test-editing-an-existing-package) to cover the change you're making. Including tests allows this PR to be merged by yourself and the owners of this module.
> 
> _**This can potentially save days of time for you!**_

No additional tests would reflect the change to the `Shipment` object.

File: types/shippo/index.d.ts [hunk 1]
```diff
@@ -471,14 +471,28 @@ declare namespace Shippo {
...
     }
 
     namespace Shipment {
         type Status = "WAITING" | "QUEUED" | "SUCCESS" | "ERROR";
+
+        // https://docs.goshippo.com/shippoapi/public-api/#tag/Shipments
+        // https://goshippo.com/docs/reference#shipments
+        // The text description of the object schema differs from the object schema
+        // below the text and in the example. The schema from the example is used here.
+        interface Message {
```
[the remark this answers is not in the document]
Comment (clalexander) [reply types/shippo/index.d.ts]: Converted `Message` to the above shape with all properties optional and removed the `Shipment.Message` type.

**Review by jlismore — APPROVED** [review]
lgtm

**Comment by clalexander** [comment]
Ready to merge

**Pull request merged**